src/index.js

```diff
@@ -670,7 +671,18 @@ const members = {
       return this.getAttribute('name');
     },
     set(name) {
-      return this.setAttribute('name', name);
+      const ret = this.__setAttribute('name', name);
+      if (!isSlotNode(this)) {
+        return ret;
+      }
+      const root = slotToRootMap.get(this);
+      const slotHasRoot = root;
```

Comment:
Does `if (root) {` not describe what's happening? I think `!!root` would be better than `slotHasRoot` but I think just using `root` is fine.